qemu/tests/virtio_fs_share_data.py

```diff
@@ -91,26 +91,32 @@ def get_viofs_exe(session):
     cmd_xfstest = params.get('cmd_xfstest')
     fs_dest_fs2 = params.get('fs_dest_fs2')
     xfstest_pkg = params.get('xfstest_pkg')
-    cmd_unpack_xfstest = params.get('cmd_unpack_xfstest')
-    cmd_yum_install = params.get('cmd_yum_install')
-    cmd_make_xfs = params.get('cmd_make_xfs')
-    cmd_setenv = params.get('cmd_setenv')
-    cmd_useradd = params.get('cmd_useradd')
+    guest_pre_command = params.get('guest_pre_command')
     fs_dest_fs1 = params.get('fs_dest_fs1')
+    cmd_get_tmpfs = params.get('cmd_get_tmpfs')
+    cmd_set_tmpfs = params.get('cmd_set_tmpfs')
+    size_mem1 = params.get('size_mem1')
 
     # xfstest-nfs config
     setup_local_nfs = params.get('setup_local_nfs')
 
-    if setup_local_nfs:
-        for fs in params.objects("filesystems"):
-            nfs_params = params.object_params(fs)
-            params["export_dir"] = nfs_params.get("export_dir")
-            params["nfs_mount_src"] = nfs_params.get("nfs_mount_src")
-            params["nfs_mount_dir"] = nfs_params.get("fs_source_dir")
-            nfs_local = nfs.Nfs(params)
-            nfs_local.setup()
-        params["start_vm"] = "yes"
-        env_process.preprocess_vm(test, params, env, params["main_vm"])
+    if cmd_xfstest:
+        # /dev/shm is the default memory-backend-file, the default value is the
+        # half of the host memory. Increase it to guest memory size to avoid crash
+        ori_tmpfs_size = process.run(cmd_get_tmpfs, shell=True).stdout_text.replace("\n", "")
+        logging.debug("original tmpfs size is %s", ori_tmpfs_size)
+        if process.run(cmd_set_tmpfs % size_mem1, shell=True).exit_status:
+            test.fail('xfstest set tmpfs failed.')
```

Comment:
```suggestion
        ori_tmpfs_size = process.run(cmd_get_tmpfs, shell=True).stdout_text.replace("\n", "")
        params["post_command"] = cmd_set_tmpfs % ori_tmpfs_size
        params["pre_command"] = cmd_set_tmpfs % size_mem1
```

We mean that set tmpfs size and recover tmpfs into pre-process and post-process like above.